src/main/java/com/auth0/IdTokenVerifier.java

```diff
@@ -0,0 +1,166 @@
+        }
+        if (!decoded.getIssuer().equals(verifyOptions.issuer)) {
+            throw new TokenValidationException(String.format("Issuer (iss) claim mismatch in the ID token, expected \"%s\", found \"%s\"", verifyOptions.issuer, decoded.getIssuer()));
+        }
+
+        if (isEmpty(decoded.getSubject())) {
+            throw new TokenValidationException("Subject (sub) claim must be a string present in the ID token");
+        }
+
+        final List<String> audience = decoded.getAudience();
+        if (audience == null) {
+            throw new TokenValidationException("Audience (aud) claim must be a string or array of strings present in the ID token");
+        }
+        if (!audience.contains(verifyOptions.audience)) {
+            throw new TokenValidationException(String.format("Audience (aud) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", verifyOptions.audience, decoded.getAudience()));
+        }
+
+        final Calendar cal = Calendar.getInstance();
+        final Date now = verifyOptions.clock != null ? verifyOptions.clock : cal.getTime();
+        final int clockSkew = verifyOptions.clockSkew != null ? verifyOptions.clockSkew : DEFAULT_CLOCK_SKEW;
+
+        if (decoded.getExpiresAt() == null) {
+            throw new TokenValidationException("Expiration Time (exp) claim must be a number present in the ID token");
+        }
+
+        cal.setTime(decoded.getExpiresAt());
+        cal.add(Calendar.SECOND, clockSkew);
+        Date expDate = cal.getTime();
+
+        if (now.after(expDate)) {
+            throw new TokenValidationException(String.format("Expiration Time (exp) claim error in the ID token; current time (%d) is after expiration time (%d)", now.getTime() / 1000, expDate.getTime() / 1000));
+        }
+
+        if (decoded.getIssuedAt() == null) {
+            throw new TokenValidationException("Issued At (iat) claim must be a number present in the ID token");
+        }
+
+        cal.setTime(decoded.getIssuedAt());
+        cal.add(Calendar.SECOND, -1 * clockSkew);
+        Date iatDate = cal.getTime();
+
+        if (now.before(iatDate)) {
+            throw new TokenValidationException(String.format("Issued At (iat) claim error in the ID token; current time (%d) is before issued at time (%d)", now.getTime() / 1000, iatDate.getTime() / 1000));
+        }
+
+
+        if (verifyOptions.nonce != null) {
+            String nonceClaim = decoded.getClaim(NONCE_CLAIM).asString();
+            if (isEmpty(nonceClaim)) {
+                throw new TokenValidationException("Nonce (nonce) claim must be a string present in the ID token");
+            }
+            if (!verifyOptions.nonce.equals(nonceClaim)) {
+                throw new TokenValidationException(String.format("Nonce (nonce) claim mismatch in the ID token; expected \"%s\", found \"%s\"", verifyOptions.nonce, nonceClaim));
+            }
+        }
+
+        if (audience.size() > 1) {
```

Comment:
As a nit, I'd do audience related checks in the same place so it's easier to track. 
I know the other check is done before this block.

Reply:
Actually it's not a problem after some thought. 
The check's purpose is different.